### DISCUSSION TITLE
How to reload mockoon-cli?

### DISCUSSION THREAD

**Discussion by eusonlito** [mockoon/mockoon#787] [Help] [answered]
**Mockoon Cli Version**
```
@mockoon/cli/2.1.0 linux-x64 node-v16.15.1
```

I'm using mockoon-cli on a server to response regular request behind a nginx proxy.

I have added the mockoon-cli as pm2 service with:

```bash
$> mockoon-cli start --data resources/mockoon.json --port 3002 --hostname 127.0.0.1 --pname api-fake
Mock started at http://127.0.0.1:3002 (pid: 12, name: mockoon-api-fake)
```

And is working fine, but I don't know how to reload on changes.

I have tried to restart or reload pm2 service with:

```bash
$> pm2 restart mockoon-api-fake
$> pm2 reload mockoon-api-fake
```

And the process is restarted on pm2 monitor, but no changes on `mockoon-cli` responses.

Also, there aren't a `mockoon-cli` restart or reload commands.

Then I have an script with a delete and create, but have no sense because is creating a new pid on every change deployed:

```bash
#!/bin/bash

pm2 del mockoon-api-fake

mockoon-cli start --data resources/mockoon.json --port 3002 --hostname 127.0.0.1 --pname api-fake

exit 0
```

How is the "right-way" to reload `mockoon-cli`?

Thanks!

**Comment by 255kb** [comment, marked as answer]
Currently, `mockoon-cli start` prepares the data (import, migrations, etc) in a new file before running a script with PM2. So, reloading using PM2 will reload the CLI instance but always with the same prepared data. 
To reload the CLI with new data, you will have to do a `mockoon-cli stop {pid|process_name}` followed by a `mockoon-cli start`.

**Reply by eusonlito** [reply]
Hi! Thanks, but I have tried also `mockook-cli stop` > `mockook-cli start` but the data is not updated.

**Reply by 255kb** [reply]
Stopping and starting the CLI with the new file passed with the `--data` flag should give new the new version of the mock. 
Do you get any error message? Or could you give me the steps to reproduce to check if there is an issue?

**Reply by eusonlito** [reply]
Sorry a lot... now is working. My fault. My problem was a node version incompatibility on first tests.
